src/test/fuzz/process_message.cpp

```diff
@@ -34,28 +34,30 @@
 
 namespace {
 const TestingSetup* g_setup;
+std::string_view LIMIT_TO_MESSAGE_TYPE{};
 } // namespace
 
-size_t& GetNumMsgTypes()
+auto& MsgTypes()
 {
-    static size_t g_num_msg_types{0};
-    return g_num_msg_types;
+    static std::set<std::string_view> g_msg_types{};
+    return g_msg_types;
 }
 #define FUZZ_TARGET_MSG(msg_type)                                            \
     struct msg_type##_Count_Before_Main {                                    \
         msg_type##_Count_Before_Main()                                       \
         {                                                                    \
-            ++GetNumMsgTypes();                                              \
+            Assert(MsgTypes().emplace(#msg_type).second);                    \
         }                                                                    \
     } const static g_##msg_type##_count_before_main;                         \
-    FUZZ_TARGET_INIT(process_message_##msg_type, initialize_process_message) \
-    {                                                                        \
-        fuzz_target(buffer, #msg_type);                                      \
-    }
 
 void initialize_process_message()
 {
-    Assert(GetNumMsgTypes() == getAllNetMessageTypes().size()); // If this fails, add or remove the message type below
+    Assert(MsgTypes().size() == getAllNetMessageTypes().size()); // If this fails, add or remove the message type below (FUZZ_TARGET_MSG)
```

Comment:
I think we can just get rid of the `FUZZ_TARGET_MSG` macro including `MsgTypes()` and use `getAllNetMessageTypes` directly?